static/mixins/polling.jsx

```diff
@@ -6,11 +6,67 @@ var PollingMixin = {
       pollingActive: true
     };
   },
+  pushNotification(){
+    let body;
+    let {app, env, number} = this.props.params;
+    const url = `/deploys/${app}/${env}/${number}`;
 
-  componentWillMount() {
+    if(this.state.task !== undefined){
+      body = `${/[^@]*/.exec(this.state.task.user.name)}'s deploy ${this.state.task.status}`;
+    }else{
+      body = `${/[^@]*/.exec(this.state.deploys[0].user.name)}'s deploy ${this.state.deploys[0].status}`;
+    }
+
+    const options = {
+      body: body,
+      icon: "../static/favicon.png"
+    }
+
+    if (!("Notification" in window)) {
+      alert("This browser does not support system notifications");
```

Comment:
Can we not `alert()` here? This should just fail silently. I'd also move this check up to the top. Otherwise, this just means that this poor person will constantly be getting spammed with alerts.